framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
Your first example (https://github.com/typetools/checker-framework/pull/634#discussion_r57753925) wasn't a valid counterexample, because it didn't involve a type refinement: the RHS was a supertype of the LHS.

To reproduce the issue, do:
- clone or create a branch of the current checker-framework, set up project
- copy the attached PolyAllVisitor.java to checker-framework/framework/tests/test-polyall/polyall
- copy the attached TypeRefinement.java to checker-framework/framework/tests/polyall
- compile the tests
- in checker-framework/framework do something like:

`javac -cp ./build:./tests/build -processor polyall.PolyAllChecker tests/polyall/TypeRefinement.java`

You should see:

```
tests/polyall/TypeRefinement.java:6: error: [polyall.h1s2.forbidden] (polyall.h1s2.forbidden)
	@H1Top Object o2 = new @H1S2 Object ();
	              ^
tests/polyall/TypeRefinement.java:6: error: [polyall.h1s2.forbidden] (polyall.h1s2.forbidden)
	@H1Top Object o2 = new @H1S2 Object ();
	                   ^
2 errors
```

The goal would be to get the error only on the explicit use of `@H1S2` not on the one that is the result of a type refinement.
Is the way we wrote the type validator not the recommended approach?

[BugReport.tar.gz](https://github.com/typetools/checker-framework/files/219724/BugReport.tar.gz)